### PR TITLE
Fix Symfony 3 compatibility

### PR DESCRIPTION
**Pull request by WouterJ**
Fixes https://github.com/KnpLabs/KnpMenuBundle/issues/291

### CODE REVIEW

**Comment by stof** [comment]
please bump the ``symfony/routing`` min dependency as this constant does not exist in 2.1 (making 2.3 is fine)

And please also fix the tests to use the constant when defining expectations of the mock (which is why tests are broken with dev deps currently as the constant value changes in 3.0)

**Comment by peterrehm** [comment]
@WouterJ Any progress on that? Some tests still seem failing...

**Comment by WouterJ** [comment]
Ready to be merged

File: composer.json [hunk 1]
```diff
@@ -25,8 +25,9 @@
     "require-dev":  {
         "pimple/pimple": "~1.0",
         "symfony/phpunit-bridge": "~2.7",
-        "silex/silex":  "~1.0",
-        "twig/twig":    "~1.16|~2.0"
+        "symfony/routing": "~2.3",
```
Comment (stof) [inline composer.json]: please allow 3.0 here too (and for the bridge), so that it would be tested against them

**Comment by peterrehm** [comment]
@WouterJ Have you had a chance to look into @stof's comment to get this merged in? It would be gread to get this merged and tagged to avoid the deprecation messages. Thanks!

**Comment by WouterJ** [comment]
Done, thanks for the ping!

**Comment by peterrehm** [comment]
@stof Can this be merged and a release tagged?

**Comment by macintoshplus** [comment]
+1

**Comment by kristijanhusak** [comment]
+1

**Comment by Garfield-fr** [comment]
:+1:

**Comment by stof** [comment]
Thank you @WouterJ.

**Pull request merged**